### ISSUE TITLE
GraphBuilder 0.1.27 rejects omitted optional outputs accepted by native checker

### ISSUE THREAD

**Issue by xadupre** [xadupre/onnx-light#4991]
## Reproduction with the published 0.1.27 wheel

`GraphBuilder` rejects a valid one-output `MaxPool` that the native checker accepts.
This blocks the native migration in xadupre/yet-another-onnx-builder#2354.

```python
from onnx_light.onnx import TensorProto, checker, helper
from onnx_light.onnx_core.graph_builder import GraphBuilder

model = helper.make_model(
    helper.make_graph(
        [helper.make_node("MaxPool", ["X"], ["Y"], kernel_shape=[2, 2])],
        "pool",
        [helper.make_tensor_value_info("X", TensorProto.FLOAT, [1, 1, 4, 4])],
        [helper.make_tensor_value_info("Y", TensorProto.FLOAT, [1, 1, 3, 3])],
    ),
    opset_imports=[helper.make_opsetid("", 21)],
)
checker.check_model(model)  # succeeds
GraphBuilder(model)  # fails
```

Actual error:

```text
graph_builder.cc:771: GraphBuilder: node 'MaxPool' was given 1 outputs
but the schema at opset 21 expects between 2 and 2.
```

Expected: the builder accepts the required first output without requiring the
optional indices output. Both importing a model and incrementally adding a node
should honor optional formal parameters.

Other failures with the same signature in the YOBX CI:

- `BatchNormalization` opset 21: one output rejected, expecting 3..3.
- `Attention` opset 24: one output rejected, expecting 4..4.
- `SoftmaxCrossEntropyLoss` opset 21: one output rejected, expecting 2..2.

Please audit native schema min/max arity computation for optional inputs and
outputs, rather than requiring callers to invent unused outputs. Add C++ backend
regressions for the valid omitted-optional-output forms and invalid arities.

Validated locally with the release asset
`onnx_light-0.1.27-cp312-cp312-manylinux_2_27_x86_64.manylinux_2_28_x86_64.whl`,
not an onnx-light source checkout.
